### PR TITLE
Fix grouped windows not being properly focused on activation

### PR DESCRIPTION
**Pull request by UserSv4**
This bug could happen if you:
1. Open two Chromium windows in a single group
2. Open some tabs in both of the windows
3. Using tabs search (Ctrl+Shift+A) try to switch to a tab in the second window from the first one.
When this happens any window focus would be lost (hyprctl activewindow starts to output "Invalid") and also the mouse cursor would become centered relatively to the second window.

### CODE REVIEW

**Comment by q234rty** [comment]
Does this fix #2675 ?

File: src/Compositor.cpp [hunk 1]
```diff
@@ -826,12 +826,16 @@ CMonitor* CCompositor::getMonitorFromOutput(wlr_output* out) {
 }
 
 void CCompositor::focusWindow(CWindow* pWindow, wlr_surface* pSurface) {
-
```
Comment (vaxerski) [inline src/Compositor.cpp:829]: why was this removed?

**Review by vaxerski — COMMENTED** [review]
romania

File: src/Compositor.cpp [hunk 2]
```diff
@@ -832,6 +832,11 @@ void CCompositor::focusWindow(CWindow* pWindow, wlr_surface* pSurface) {
         return;
     }
 
+    if (pWindow && pWindow->isHidden() && pWindow->m_sGroupData.pNextWindow) {
```
[the remark this answers is not in the document]
Comment (vaxerski) [reply src/Compositor.cpp:835]: oh fuck right, hidden() windows don't count as validMapped. My bad.

**Review by vaxerski — APPROVED** [review]
computer
